datasets/fr/assemblee/crawler.py

```diff
@@ -0,0 +1,129 @@
+import re
+from typing import Any, Dict, Iterator, List
+from zipfile import ZipFile
+
+from zavod import Context, Entity
+from zavod import helpers as h
+from zavod.logic.pep import categorise
+
+REGEX_PATH = re.compile(r"^json/acteur/")
+
+
+def crawl_collabos(
+    context, person: Entity, uid: str, mandat: Dict[str, Any]
+) -> Iterator[Entity]:
+    """Add staff (parliamentry collaborators) as associates."""
+    collabos = mandat.pop("collaborateurs")["collaborateur"]
+    if not isinstance(collabos, list):
+        collabos = [collabos]
+    for i, c in enumerate(collabos):
+        collabo = context.make("Person")
+        collabo.id = context.make_slug(uid, f"collabo{i}")
+        h.apply_name(
+            collabo,
+            prefix=c.pop("qualite"),
+            first_name=c.pop("prenom"),
+            last_name=c.pop("nom"),
+        )
+        collabo.set("topics", "role.rca")
+        context.audit_data(c, ["dateDebut", "dateFin"])
+        yield collabo
+
+        link = context.make("Associate")
+        link.id = context.make_slug(uid, f"associate-collabo{i}")
+        link.set("person", person)
+        link.set("associate", collabo)
+        link.set("relationship", "collaborateur")
+        yield link
+
+
+def crawl_acteur(context, data: Dict[str, Any]):
+    """Extract MNA information from JSON."""
+    acteur = data.pop("acteur")
+    context.audit_data(data, ["@xmlns", "profession"])
+
+    person = context.make("Person")
+    # Member UID
+    uid = acteur.pop("uid")
+    uid_text = uid.pop("#text")
+    person.id = context.make_slug(uid_text)
+    context.audit_data(uid, ["@xmlns:xsi", "@xsi:type"])
+
+    # Name and DOB
+    ec = acteur.pop("etatCivil")
+    ident = ec.pop("ident")
+    h.apply_name(
+        person,
+        prefix=ident.pop("civ"),
+        first_name=ident.pop("prenom"),
+        last_name=ident.pop("nom"),
+        lang="fra",
+    )
+    context.audit_data(ident, ["alpha", "trigramme"])
+    dob = ec.pop("infoNaissance")
+    person.set("birthDate", h.parse_date(dob.pop("dateNais"), ["%Y-%m-%d"]))
+    person.set("birthPlace", dob.pop("villeNais"))
+    person.set("birthCountry", dob.pop("paysNais"))
+    # Ignore birth departement for now
+    context.audit_data(dob, ["depNais"])
+    # Assume current MNAs are not dead
+    context.audit_data(ec, "dateDeces")
+
+    # We'll include this URL in the data as it's quite useful
+    hatvp = acteur.pop("uri_hatvp")
+    person.set("sourceUrl", hatvp)
+
+    # Addresses and phone numbers are available because transparence!
+    # But we do not include them because vie privée!
+    acteur.pop("adresses")
+
+    # Now make position etc
+    position = h.make_position(
+        context,
+        "member of the French National Assembly",
+        country="fr",
+    )
+    categorisation = categorise(context, position, is_pep=True)
+    if not categorisation.is_pep:
+        return
+    # Occupancy data (there are many mandats but we only care about AN)
```

Comment:
Verify, is this true?  Do we  care about the various committees, extraparliamentry jamborees, etc?

Reply:
if AN stands for Assemblee Nationale then yes - the other mandates appear to be membership of commissions, and other bits and bobs which I don't think tell us enough extra about their PEPishness to bother - we don't normally include committee membership, for example. Enough to know they're a legislator.

I did come across one deputy judge position, but since that's not the focus of this dataset and we can't expect to find close to the full set of judicial PEPs here, let's not bother.